src/new-client/src/App/LiveRates/Tiles/ExecutionResponse/Response.tsx

```diff
@@ -76,51 +77,69 @@ const buildTradeMessage = (
   </TradeMessageDiv>
 )
 
-const ResponseContainer = (props: any) => {
+export const ExecutionResponse = () => {
   const symbol = useContext(SymbolContext)
-  const { currencyPair, id, tradeId } = useExecution(symbol)
+  const {
+    currencyPair,
+    id,
+    tradeId,
+    direction,
+    notional,
+    spotRate,
+    valueDate,
+    status,
+  } = useExecution(symbol)
   const baseTerm = useBaseTerm(symbol)
-
+  const { base, terms } = useCurrencyPair(symbol)
+  const successOrReject =
+    status === ExecutionStatus.Done || status === ExecutionStatus.Rejected
   const closeAlert = () => {
     onExecutionDismiss({ currencyPair, id })
   }
 
-  return (
-    <ExecutionStatusAlertContainer status={props.status}>
-      <CurrencyPairDiv>
-        {props.status === ExecutionStatus.Done ? (
-          <AlignedCheck />
-        ) : (
-          <AlignedTriangle />
+  return status !== ExecutionStatus.Ready ? (
```

Comment:
We're removing all duplication, but we're losing a lot of readability, IMO.   Feel like the best mixed strategy would be inside main block:

```
<ExecutionStatusAlertContainer status={status}>
{
   status === ExecutionStatus.Done && <Done /> ||
   status === ExecutionStatus.Rejected && <Rejected /> ||
   status === ExecutionStatus.ExecutionStatus.TakingTooLong && <TakingTooLong /> ||
   status === ExecutionStatus.RequestTimeout && <RequestTimeOut />
</ExecutionStatusAlertContainer>

// Done, Rejected, TakingTooLong, RequestTimeOut then compose their shared UI as sub-components
```

What's your view @josepot ?